pkg/inject/inject.go

```diff
@@ -539,6 +546,32 @@ func (conf *ResourceConfig) serviceAccountVolumeMount() *corev1.VolumeMount {
 	return nil
 }
 
+func (conf *ResourceConfig) trace() *charts.Trace {
+	var (
+		svcAddr    = conf.getOverride(k8s.ProxyTraceCollectorSvcAddr)
+		svcAccount = conf.getOverride(k8s.ProxyTraceCollectorSvcAccount)
+	)
+
+	if svcAddr == "" {
+		return nil
+	}
+
+	splits := strings.Split(svcAddr, ".")
+	if len(splits) <= 1 {
+		log.Errorf("fail to enable tracing due to malformed collector service address %s. Expected service address to use the <svc-name>:<ns> format", svcAddr)
```

Comment:
Do we want to support a single segment service name if the service is in the same namespace as the pod?

Reply:
Good point. I forgot that the `.ns` suffix is optional if the `Service` is in the same namespace.

Reply:
Since the namespace that will be appended to the service account below is derived from the service address, a single segment service name isn't going to work. We will need to add this to the docs.

Reply:
if the service namespace isn't provided, isn't it assumed to be the same as the pod?  i.e. if a pod in the `foo` namespace specifies the `bar` service, we would assume that means the `bar` service in the `foo` namespace and also use `foo` as the namespace for the service account?